### DISCUSSION TITLE
LXP register for new Charge Enabled feature

### DISCUSSION THREAD

**Discussion by timbotwenty** [celsworth/lxp-bridge#268] [General]
Hi, Lux have recently (8th May 2024) added/enabled the Charge Last feature on the Lux portal. Does anyone know the register for this so it can be set via lxp-bridge?

I have looked in LXP-REGISTERS but can't see it there.

Note this is an enable/disable button, that is visible on the web portal (not app) Remote Set page. When enabled it allows the inverter to export all PV up to the export limit (eg 3.68kW for standard UK DNO), and then send the remainder to the battery once export limit is reached, and thus prevent clipping. 

Thanks in advance.

**Comment by celsworth** [comment]
It's not a register I recognise. If it wasn't in the (now starting to get a bit old) documentation I got the registers from then it won't be in the docs in this repo either.

Probably the best way to get this is to enable debug logging in lxp-bridge, then press the button :) See what lxp-bridge logs. We should be able to see what register gets set because even though lxp-bridge can't see the original packet sent from Luxpower to the inverter, we *can* see the reply the inverter sends (it gets broadcast to all connected clients). This reply will have the register and new register value in, hopefully.

**Reply by timbotwenty** [reply]
Thanks Chris. Confession time, I am still on octolux - does that have the same feature (debug logging)? At a guess adding a log statement inside read_reply(pkt) (in lux_controller.rb) is what I would need? Thx

**Reply by celsworth** [reply]
Oh wow, haven't looked at octolux for a year or more ;)

I think read_reply might only be called when we're waiting for a response to a command we sent to the inverter.. Perhaps you want to be looking more around here:
https://github.com/celsworth/octolux/blob/31d7d615a7e93562c0aefe1888765c6212b4254b/lib/lux_listener.rb#L36

Hopefully that sees every incoming packet from the inverter, not just ones we're looking for..

**Reply by timbotwenty** [reply]
Thanks, I will have a play with the listener
